### ISSUE TITLE
3D secure 2 data are not propagated

### ISSUE THREAD

**Issue by piotrchowaniec** [globalpayments/java-sdk#38]
I'm trying to create a charge with 3D secure 2 data like in the [docs (see Authorization)](https://developer.globalpay.com/docs/sdk/3DS2#):

```
// Create the card object
CreditCardData card = new CreditCardData();
card.setNumber("4012001038488884");
card.setExpMonth(12);
card.setExpYear(2025);
card.setCvn("123");
card.setCardHolderName("James Mason");

// Add obtained 3D Secure 2 authentication data
ThreeDSecure threeDSecureData = new ThreeDSecure();
threeDSecureData.setAuthenticationValue("ODQzNjgwNjU0ZjM3N2JmYTg0NTM=");
threeDSecureData.setDirectoryServerTransactionId("c272b04f-6e7b-43a2-bb78-90f4fb94aa25");
threeDSecureData.setEci("05");
threeDSecureData.setMessageVersion("2.1.0");

// Add the 3D Secure 2 data to the card object
card.setThreeDSecure(threeDSecureData);

Transaction response = null;

try {
   response = card.charge(new BigDecimal("10.01"))
         .withCurrency("USD")
         .execute();
}

catch (ApiException exec) {
   // TODO: add your error handling here
}
```

But the request builder omits all 3DS related fields except `exempt_status`, and the actual body of the request is like this:
```
...
"authentication": {
      "three_ds": [
        {}
      ]
    }
...
```

Problem is in the `GpApiAuthorizationRequestBuilder` (from lines 289) implementation.
Is there any reason why this is implemented this way? Why other parameters are not included in the request? Is it possible to fix this issue?

**Comment by piotrchowaniec** [comment]
Another problem connected with this one is that in generated json `three_ds` is an array, but should be an object. So it should be something like this:
```
"three_ds": {
        "eci": "5",
        "ds_trans_reference": "ref",
        "value": "value",
        "message_version": "2.1.0"
      }
```

**Comment by piotrchowaniec** [comment]
It was fixed in v11.1.23
